File: Days.32/C++/1.Prefix-to-Postfix.cpp
// Prefix ----> Postfix
#include<iostream>
#include<vector>
using namespace std;


string PostfixtoPrefix(string &str)
{
    vector<string> op;
    string a,b,temp;
    for (int i=str.size()-1; i>=0; i--)
    {
        if (str[i]=='+' or str[i]=='-' or str[i]=='*' or str[i]=='/' or str[i]=='^')
        {
            a=op.back();
            op.pop_back();
            b=op.back();
            op.pop_back();
            op.push_back(a+b+str[i]);
        }
        else
        {
            temp=str[i];
            op.push_back(temp);
        }
    }
    return op.back();
}


int main()
{
    string str;
    cin>>str;
    string ans=PostfixtoPrefix(str);
    cout<<ans<<"\n";
    return 0;
}

File: Days.32/Python/1.Prefix-to-Postfix.py
# Prefix ----> Postfix


def PrefixtoPostfix(str):
    op=[]
    for i in range(len(str)-1,-1,-1):
        if str[i]=='+' or str[i]=='-' or str[i]=='*' or str[i]=='/' or str[i]=='^':
            a=op.pop()
            b=op.pop()
            op.append(a+b+str[i])
        else:
            op.append(str[i])
    
    return op[-1]
            


def main():
    str=input()
    ans=PrefixtoPostfix(str)
    print(ans)



main()

